robot.py

```diff
@@ -333,7 +359,63 @@ def teleopInit(self):
         self.drivetrain.disableVoltageCompensation()
         return
 
+    def checkTeamColorAndFieldPosition(self):
+
+        #if the bot team color selection on elastic has changed
+        selectedTeam = self.elastic.getSelectedTeam()
+        if selectedTeam != self.ELASTIC_BOT_COLOR: #checks if selection is different
+            self.ELASTIC_BOT_COLOR = selectedTeam
+            if selectedTeam == "RED":
+                teamColorIsRed = True      #will be passed to override/update function
+            elif selectedTeam == "BLUE":
+                teamColorIsRed = False
+
+            teamGyroAdjustment, teamMoveAdjustment = self.updateTeamColor(teamColorIsRed)
+            self.swervometer.updateTeam(teamGyroAdjustment, teamMoveAdjustment)
+
+        #if the bot starting position selection on elastic has changed
+        selectedPosition = self.elastic.getSelectedPosition()
+        if selectedPosition != self.ELASTIC_BOT_POSITION:
+            self.ELASTIC_BOT_POSITION = selectedPosition
+
+            positions = self.savedSwervometerConfig['ALL_ACTIVE_POSITIONS']
+            for i in positions:
+                if selectedPosition == i:
+                    fieldStartPosition = selectedPosition
+                else:
+                    None
+
+            starting_position_x, starting_position_y, starting_angle = self.updateFieldStartPosition(fieldStartPosition) #one letter splits into 3 values
+            self.swervometer.updateFieldStartPosition(starting_position_x, starting_position_y, starting_angle) #3 values get sent to swervometer
+            
+        #if the bot auton plan selection on elastic has changed
+        selectedAuton = self.elastic.getSelectedAuton()
+        if selectedAuton != self.ELASTIC_BOT_AUTON:
+            self.ELASTIC_BOT_AUTON = selectedAuton
+            self.taskListName = selectedAuton #Auton plan is already selected in autonomousInit
+
     def robotPeriodic(self):
+        
+        if (self.elastic):
+            
+            #updates the variables and displays on elastic field
+            self.checkTeamColorAndFieldPosition()
+            
+            #indicates if a note is loaded inside the robot (beam is broken)
+            self.elastic.updateBeamDisplay(self.mechanism.indexBeamBroken())
+
+            """ This does not work, isConnected() isnt a function and there is nothing on documentation
+            #the goal is to create an indicator that displays if the controllers are connected (individually)
+            isDriverConnected = False
+            isOperatorConnected = False
+            if self.driver:
+                isDriverConnected = wpilib.DriverStation.getJoystickIsXbox(0) #test this
+                if isDriverConnected:
+                    isDriverConnected = True
+            print(isDriverConnected, isOperatorConnected)
+            self.elastic.updateControllerConnectedDisplay(isDriverConnected, isOperatorConnected)
```

Comment:
I'd try `self.operator.xboxController.isConnected()` and the same for `self.driver`. Note that you will probably have to confirm that the Controllers (`self.operator`, `self.driver`) have already been initialized.